### PR TITLE
Provide source code documentation

### PR DESCRIPTION
**Pull request by falk-werner**
This pull request provides automatically created source code documentation.

The documentation ist created by any push to the main branch and is pushed to Github pages:  
[https://falk-werner.github.io/note.py](https://falk-werner.github.io/note.py)
By doing so, the documentation will always be up to date.

The documentation can also be created locally by running the script `doc/create_doc`. Therefore, [Sphinx](https://www.sphinx-doc.org/en/master/) and [Read the Docs Sphinx Theme](https://sphinx-rtd-theme.readthedocs.io/en/stable/) must be installed. A `requirements.txt` is provided in the `doc` directory which provides these packages via pip:

    pip3 install -r doc/requirements.txt

### CODE REVIEW

File: note.py [hunk 1]
```diff
@@ -159,27 +170,55 @@ def __note_filename(self, name):
         return os.path.join(self.__notespath, name, "note.md")
 
     def geometry(self, geometry=None):
-        """Returns the geometry (size) of the application window"""
+        """
+        Returns and optionally sets the geometry (size)
```
Comment (hmlampe) [inline note.py:174]:
```suggestion
        Returns and optionally sets the geometry
```
it saves size and position

File: note.py [hunk 2]
```diff
@@ -159,27 +170,55 @@ def __note_filename(self, name):
...
+        of the application window.
+
+        :param geometry: Optional geometry of application window (Default: None).
+        :type  geometry: str | None
+
+        :return: Configured geometry (size) of the application window.
```
Comment (hmlampe) [inline note.py:180]:
```suggestion
        :return: Configured geometry (of the application window.
```
if position is changed once, it is saved, too

File: note.py [hunk 3]
```diff
@@ -197,34 +245,61 @@ def read_note(self, name):
...
+        """
         note_path = self.note_path(name)
         if os.path.isdir(note_path):
             shutil.rmtree(note_path)
 
     def screenshot(self, name):
-        """Takes a screenshot and returns it's filename."""
+        """Takes a screenshot and returns it's filename.
+
+        :param name: Name of the not the screenshot is assigned to.
```
Comment (hmlampe) [inline note.py:285]:
```suggestion
        :param name: Name of the note the screenshot is assigned to.
```

File: note.py [hunk 4]
```diff
@@ -294,21 +412,38 @@ def delete(self):
         self.__parent.note_changed()
 
     def screenshot(self):
-        """Takes a screenshot and return the filename."""
+        """Takes a screenshot and return the filename.
```
Comment (hmlampe) [inline note.py:415]:
```suggestion
        """Takes a screenshot and returns the filename.
```

File: note.py [hunk 5]
```diff
@@ -378,23 +536,43 @@ def __init__(self, persistence=Persistence()):
...
+        :return: Name of the app.
+        :rtype: str
+        """
         return self.__name
 
     def get_geometry(self):
-        """Returns the size of the main window."""
+        """Returns the configured geometry (size) of the main window.
+
+        :return: Configured geometry (size) of the main window.
```
Comment (hmlampe) [inline note.py:549]:
```suggestion
        :return: Configured geometry of the main window.
```

File: note.py [hunk 6]
```diff
@@ -378,23 +536,43 @@ def __init__(self, persistence=Persistence()):
...
+        :rtype: str
+        """
         return self.__geometry
 
     def set_geometry(self, geometry):
-        """Sets the size of the main window."""
+        """Sets the geometry (size) of the main window.
```
Comment (hmlampe) [inline note.py:555]:
```suggestion
        """Sets the geometry of the main window.
```

File: note.py [hunk 7]
```diff
@@ -378,23 +536,43 @@ def __init__(self, persistence=Persistence()):
...
+
+        :param geometry: Geometry (size) of the main windows.
```
Comment (hmlampe) [inline note.py:557]:
```suggestion
        :param geometry: Geometry of the main windows.
```

File: note.py (hunk 2 above)
Comment (falk-werner) [reply note.py]: fixed

**Review by hmlampe — APPROVED** [review]

**Pull request merged**